fastly/tls_subscription.go

```diff
@@ -0,0 +1,224 @@
+package fastly
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/google/jsonapi"
+)
+
+// TLSSubscription represents a managed TLS certificate
+type TLSSubscription struct {
+	ID                   string                        `jsonapi:"primary,tls_subscription"`
+	CertificateAuthority string                        `jsonapi:"attr,certificate_authority"`
+	State                string                        `jsonapi:"attr,state"`
+	CreatedAt            *time.Time                    `jsonapi:"attr,created_at,iso8601"`
+	UpdatedAt            *time.Time                    `jsonapi:"attr,updated_at,iso8601"`
+	Configuration        *TLSConfiguration             `jsonapi:"relation,tls_configuration"`
+	CommonName           *TLSDomain                    `jsonapi:"relation,common_name"`
+	Domains              []*TLSDomain                  `jsonapi:"relation,tls_domains"`
+	Certificates         []*TLSSubscriptionCertificate `jsonapi:"relation,tls_certificates"`
+	Authorizations       []*TLSAuthorizations          `jsonapi:"relation,tls_authorizations"`
+}
+
+type TLSSubscriptionCertificate struct {
+	ID string `jsonapi:"primary,tls_certificate"`
+}
+
+type TLSAuthorizations struct {
```

Comment:
Am I right in thinking this isn't documented yet (along with TLSChallenge)?

Reply:
I think so, although I can add some comments regardless if you think that'd be helpful?

Reply:
Yes please 🙂

Reply:
I've added a comment to this and TLSChallenge -- no official docs to go off of but hopefully it gives an idea what it does 👍🏼